libraries/GCS_MAVLink/GCS_Param.cpp

```diff
@@ -375,6 +374,10 @@ void GCS_MAVLINK::param_io_timer(void)
 {
     struct pending_param_request req;
 
+    // this is mostly a no-op, but doing this here means we won't
+    // block the main thread counting parameters (~30ms on PH)
+    AP_Param::count_parameters();
```

Comment:
its being called from the IO thread, which is the right place for slow calls, and its only 30ms once per change in parameter tree shape.